src/wallet/api.py

```diff
@@ -29,10 +29,10 @@ def set_balance(owner_id, new_balance, reference=None):
     wallet, old_balance = get_balance(owner_id, new_balance.currency)
     difference = new_balance - old_balance
     if difference.amount < 0:
-        withdraw(owner_id, -difference, reference)
+        return withdraw(owner_id, -difference, reference), difference
```

Comment:
Second return value should be `-difference` to become identical to the object's `amount`